src/main/java/org/prebid/server/bidder/flipp/FlippBidder.java

```diff
@@ -0,0 +1,278 @@
+        if (StringUtils.isNoneEmpty(userId)) {
+            return userId;
+        }
+
+        return Optional.ofNullable(extImpFlipp.getUserKey())
+                .filter(StringUtils::isNotEmpty)
+                .orElseGet(() -> UUID.randomUUID().toString());
+    }
+
+    private HttpRequest<CampaignRequestBody> createRequest(BidRequest bidRequest,
+                                                           CampaignRequestBody campaignRequestBody) {
+        return HttpRequest.<CampaignRequestBody>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(updateHeaders(bidRequest.getDevice()))
+                .body(mapper.encodeToBytes(campaignRequestBody))
+                .payload(campaignRequestBody)
+                .build();
+    }
+
+    private static MultiMap updateHeaders(Device device) {
+        return Optional.of(device)
+                .map(Device::getUa)
+                .map(ua -> HttpUtil.headers().add(HttpUtil.USER_AGENT_HEADER, ua))
+                .orElseGet(HttpUtil::headers);
+    }
+
+    @Override
+    public final Result<List<BidderBid>> makeBids(BidderCall<CampaignRequestBody> httpCall, BidRequest bidRequest) {
+        try {
+            final CampaignResponseBody campaignResponseBody =
+                    mapper.decodeValue(httpCall.getResponse().getBody(), CampaignResponseBody.class);
+            return Result.withValues(extractInline(campaignResponseBody, bidRequest));
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private static List<BidderBid> extractInline(CampaignResponseBody campaignResponseBody, BidRequest bidRequest) {
+        final List<Inline> inlines = Optional.ofNullable(campaignResponseBody)
+                .map(CampaignResponseBody::getDecisions)
+                .map(Decisions::getInline)
+                .orElse(null);
+        if (CollectionUtils.isEmpty(inlines)) {
+            return Collections.emptyList();
+        }
+
+        return bidsFromResponse(inlines, bidRequest);
+    }
+
+    private static List<BidderBid> bidsFromResponse(List<Inline> inlines, BidRequest bidRequest) {
+        final List<BidderBid> bidderBids = new ArrayList<>();
+
+        for (Imp imp : bidRequest.getImp()) {
+            for (Inline inline : inlines) {
+                final String requestId = ObjectUtil.getIfNotNull(
+                        ObjectUtil.getIfNotNull(inline, Inline::getPrebid), Prebid::getRequestId);
+                if (Objects.equals(requestId, imp.getId())) {
+                    bidderBids.add(BidderBid.of(constructBid(inline, imp.getId()), DEFAULT_BID_TYPE, DEFAULT_CURRENCY));
+                }
+            }
+        }
+
+        return bidderBids;
+    }
+
+    private static Bid constructBid(Inline inline, String impId) {
+        final Prebid prebid = ObjectUtil.getIfNotNull(inline, Inline::getPrebid);
```

Comment:
Here you are not sure if inline will be present at all, but later you are blindly using *potentially* null object's methods. So, either remove this `getIfNotNull` call, or do nullability everywhere.

UPD. Checked place, where `constructBid` is called. Inline can't be null since you are calling it if and only if `inline.prebid.requestId` is equal to `imp.id`, so just remove `getIfNotNull` call.